### PR TITLE
[Web UI] Create AppLayout component

### LINKED ISSUE #1372: [Web UI] AppLayout component
Create a new `AppLayout` component to takeover the responsibility of `AppFrame` and `AppContent`.

`/src/component/layout/AppLayout` will implement the outermost layout of the page, taking `topBar`, `content`, and `quickNav` as props.

`/src/component/layout/` will serve as a location for other "layout" components to live (`SidebarLayout`, `FullPageLayout` etc.)

### PR DESCRIPTION
**Pull request by 10xjs**
## What is this change?

AppLayout defines the outermost structure common to all pages, serving as a foundation for forthcoming notification toasts and banners.

## Why is this change necessary?

Closes #1372

## Does your change need a Changelog entry?

No

## Do you need clarification on anything?

No

## Were there any complications while making this change?

No

## Have you reviewed and updated the documentation for this change? Is new documentation required?

No

### CODE REVIEW

File: dashboard/src/components/AppBar.js [hunk 1]
```diff
@@ -9,34 +9,38 @@ import { withStyles } from "@material-ui/core/styles";
...
 
       ${EnvironmentLabel.fragments.environment}
+      ${Drawer.fragments.environment}
     `,
   };
 
   static styles = theme => ({
-    container: {
-      paddingTop: "env(safe-area-inset-top)",
-      backgroundColor: theme.palette.primary.dark,
```
Comment (jamesdphillips) [inline dashboard/src/components/AppBar.js]: This padding was for devices with a notch, I think looks like it's been lost to the re-organization.

File: dashboard/src/components/AppLayout.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    flexDirection: "column",
+    alignItems: "stretch",
+
+    marginLeft: "auto",
+    marginRight: "auto",
+
+    maxWidth: 1224,
+
+    paddingLeft: 8,
+    paddingRight: 8,
```
Comment (jamesdphillips) [inline dashboard/src/components/AppLayout.js]: Why not use the theme's spacing unit here? It's nice to avoid magic numbers when we can.

Comment (jamesdphillips) [reply dashboard/src/components/AppLayout.js]: Between this and removing the padding prop from the `Content` component, we've lost a little in translation. 

Previously we have no padding in the content container on narrower viewports, this allowed lists and other elements to expand to the edge of the screen matching platform conventions.

File: dashboard/src/components/AppLayout.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+  handleTopBarResize = rect => {
+    this.setState(state => {
+      if (state.topBarHeight === rect.height) {
+        return null;
+      }
+
+      return { topBarHeight: rect.height };
+    });
+  };
```
Comment (jamesdphillips) [inline dashboard/src/components/AppLayout.js]: neat solution 👍

File: dashboard/src/components/AppLayout.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+            <div className={classes.topBarObserver}>
+              <ResizeObserver onResize={this.handleTopBarResize} />
+              {topBar}
+            </div>
+            <div className={classes.quickNavContainer}>
+              <div className={classes.quickNav}>{quickNav}</div>
+            </div>
+            <div className={classes.alertContainer}>
+              <div className={classes.alert}>{alert}</div>
+            </div>
```
Comment (jamesdphillips) [inline dashboard/src/components/AppLayout.js]: Where is this being used?

File: dashboard/src/components/AppBar.js (hunk 1 above)
Comment (10xjs) [reply dashboard/src/components/AppBar.js]: My mistake. I Thought `env(safe-area-inset-top)` was broken functionality in the style build system. I can revert this.

**Pull request merged**